src/credix_lp/depository.ts

```diff
@@ -0,0 +1,286 @@
+      throw new Error('Could not read credixGlobalMarketState');
+    }
+    // Then we read the content of the credixProgramState onchain
+    const credixProgramStateData =
+      await credixProgram.account.programState.fetchNullable(
+        credixProgramState
+      );
+    if (!credixProgramStateData) {
+      throw new Error('Could not read credixProgramState');
+    }
+
+    // We now have all the data we need from the onchain credix configuration
+    const credixSharesMint = credixGlobalMarketStateData.lpTokenMint;
+    const credixTreasuryCollateral =
+      credixGlobalMarketStateData.treasuryPoolTokenAccount;
+    const credixMultisigKey = credixProgramStateData.credixMultisigKey;
+    const credixMultisigCollateral = await this.findCredixMultisigCollateral(
+      credixMultisigKey,
+      collateralMint
+    );
+
+    // Then we can find the depository
+    const [depository] = PublicKey.findProgramAddressSync(
+      [
+        Buffer.from(CREDIX_LP_DEPOSITORY_NAMESPACE),
+        credixGlobalMarketState.toBuffer(),
+        collateralMint.toBuffer(),
+      ],
+      uxdProgramId
+    );
+
+    // Then read the collateral mint
+    const collateralToken = new Token(
+      connection,
+      collateralMint,
+      TOKEN_PROGRAM_ID,
+      null as unknown as Signer
+    );
+    const collateralInfo = await collateralToken.getMintInfo();
+    if (!collateralInfo) {
+      throw new Error('Cannot find the collateral mint');
+    }
+    const collateralDecimals = collateralInfo.decimals;
+
+    // Then derive remaining credix internal accounts from other PDAs
+    const credixSigningAuthority = await this.findCredixSigningAuthority(
+      credixGlobalMarketState,
+      credixProgramId
+    );
+    const credixLiquidityCollateral = await this.findCredixLiquidityCollateral(
+      credixSigningAuthority,
+      collateralMint
+    );
+    const credixPass = await this.findCredixPass(
+      credixGlobalMarketState,
+      depository,
+      credixProgramId
+    );
+
+    // Then generate the depository token accounts
+    const depositoryCollateral = await this.findDepositoryCollateralAddress(
+      depository,
+      collateralMint
+    );
+    const depositoryShares = await this.findDepositorySharesAddress(
+      depository,
+      credixSharesMint
+    );
+
+    // Done
+    return new CredixLpDepository(
+      depository,
+      collateralMint,
+      collateralDecimals,
+      collateralSymbol,
+      depositoryCollateral,
+      depositoryShares,
+      credixProgramState,
+      credixGlobalMarketState,
+      credixSigningAuthority,
+      credixLiquidityCollateral,
+      credixSharesMint,
+      credixPass,
+      credixTreasuryCollateral,
+      credixMultisigKey,
+      credixMultisigCollateral,
+      credixProgramId
+    );
+  }
+
+  private static async findDepositoryCollateralAddress(
+    depository: PublicKey,
+    collateralMint: PublicKey
+  ) {
+    return findATAAddrSync(depository, collateralMint)[0];
+  }
```

Comment:
no need for the method to be `async` if it uses `findATAAddrSync`

Reply:
nice catch